### PR TITLE
[LM-11488] Added DM feed

### PR DESCRIPTION
**Pull request by pushpendralike**
## Description

**Features**
- Added DM Feed

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Hotfix (fixes a breaking changes)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Other, please describe:

### Author Checklist

- [x] Assignee added
- [x] Reviewer added
- [x] Description added
- [ ] Unit test cases added

### Reviewer Checklist

- [x] PR title starts with Jira ticket number prefix
- [ ] If hotfix, Add Hotfix as prefix in PR title
- [x] Label added
- [x] Commit messages follows standard format (added, changed, removed, code commented and chore)
- [ ] Description about the Feature/ Bugs added

### CODE REVIEW

File: LMChatCore_iOS/LMChatCore_iOS/Source/Module/DirectMessageFeed/LMChatDMFeedViewController.swift [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+        button.titleLabel?.font = Appearance.shared.fonts.textFont2
+        button.tintColor = .white
+        button.backgroundColor = Appearance.shared.colors.linkColor
+        return button
+    }()
+    
+    open override func viewDidLoad() {
+        super.viewDidLoad()
+        // Do any additional setup after loading the view.
+        setupViews()
```
Comment (devansh-mohata) [inline LMChatCore_iOS/LMChatCore_iOS/Source/Module/DirectMessageFeed/LMChatDMFeedViewController.swift:49]: no need to call it, it is called in superclass `LMViewController` in `loadView()`

File: LMChatCore_iOS/LMChatCore_iOS/Source/Module/DirectMessageFeed/LMChatDMFeedViewController.swift [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+        }
+        viewModel?.getChatrooms()
+        viewModel?.syncChatroom()
+        viewModel?.checkDMStatus()
+        profileIcon.kf.setImage(with: URL(string: viewModel?.memberProfile?.imageUrl ?? ""), placeholder: UIImage.generateLetterImage(name: viewModel?.memberProfile?.name?.components(separatedBy: " ").first ?? ""))
+    }
+    
+    // MARK: setupViews
+    open override func setupViews() {
+        self.view.addSubview(feedListView)
```
Comment (devansh-mohata) [inline LMChatCore_iOS/LMChatCore_iOS/Source/Module/DirectMessageFeed/LMChatDMFeedViewController.swift:68]: super call is missing

File: LMChatCore_iOS/LMChatCore_iOS/Source/Module/DirectMessageFeed/LMChatDMFeedViewController.swift [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+        self.view.addSubview(newDMFabButton)
+        setupRightItemBars()
+    }
+    
+    // MARK: setupLayouts
+    open override func setupLayouts() {
+        NSLayoutConstraint.activate([
```
Comment (devansh-mohata) [inline LMChatCore_iOS/LMChatCore_iOS/Source/Module/DirectMessageFeed/LMChatDMFeedViewController.swift:75]: Use extension written functions

File: LMChatCore_iOS/LMChatCore_iOS/Source/Module/DirectMessageParticipant/LMChatMemberListViewController.swift [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+        let view = LMUIComponents.shared.participantListView.init().translatesAutoresizingMaskIntoConstraints()
+        view.backgroundColor = .systemGroupedBackground
+        view.delegate = self
+        return view
+    }()
+    
+    
+    // MARK: setupViews
+    open override func setupViews() {
+        self.view.addSubview(memberCountsLabel)
```
Comment (devansh-mohata) [inline LMChatCore_iOS/LMChatCore_iOS/Source/Module/DirectMessageParticipant/LMChatMemberListViewController.swift:36]: super call missing

File: LMChatCore_iOS/LMChatCore_iOS/Source/Module/DirectMessageParticipant/LMChatMemberListViewController.swift [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+        containerView.reloadList()
+        
+        var subCount: String? = nil
+        
+        if let count = viewModel?.totalParticipantCount,
+           count != 0 {
+            subCount = "\(count) members"
+        }
+        memberCountsLabel.text = subCount
+//        setNavigationTitleAndSubtitle(with: Constants.shared.strings.sendDMToTitle, subtitle: subCount)
```
Comment (devansh-mohata) [inline LMChatCore_iOS/LMChatCore_iOS/Source/Module/DirectMessageParticipant/LMChatMemberListViewController.swift:86]: remove if unused

File: LMChatCore_iOS/LMChatCore_iOS/Source/Module/LMChatHomeFeed/LMChatFeedViewController.swift [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+        didSet {
+            segmentControl.selectedSegmentIndex = currentPageIndex
+        }
+    }
+    
+    open override func viewDidLoad() {
+        super.viewDidLoad()
+        setupViews()
+        setupLayouts()
+        setupAppearance()
```
Comment (devansh-mohata) [inline LMChatCore_iOS/LMChatCore_iOS/Source/Module/LMChatHomeFeed/LMChatFeedViewController.swift:55]: no need to call these functions

File: LikeMindsChatCore.podspec [hunk 7]
```diff
@@ -20,6 +20,6 @@ Pod::Spec.new do |spec|
   spec.dependency "AWSCognito"
   spec.dependency "AWSS3"
   spec.dependency 'Giphy'
-  spec.dependency 'LikeMindsChatData'
+  spec.dependency 'LikeMindsChat'
```
Comment (devansh-mohata) [inline LikeMindsChatCore.podspec:23]: revert back to LikeMindsChatData

**Review by devansh-mohata — APPROVED** [review]

File: LikeMindsChatCore.podspec (hunk 7 above)
Comment (pushpendralike) [reply LikeMindsChatCore.podspec:23]: Done

File: LMChatCore_iOS/LMChatCore_iOS/Source/Module/LMChatHomeFeed/LMChatFeedViewController.swift (hunk 6 above)
Comment (pushpendralike) [reply LMChatCore_iOS/LMChatCore_iOS/Source/Module/LMChatHomeFeed/LMChatFeedViewController.swift:55]: Done

File: LMChatCore_iOS/LMChatCore_iOS/Source/Module/DirectMessageParticipant/LMChatMemberListViewController.swift (hunk 5 above)
Comment (pushpendralike) [reply LMChatCore_iOS/LMChatCore_iOS/Source/Module/DirectMessageParticipant/LMChatMemberListViewController.swift:86]: Done

File: LMChatCore_iOS/LMChatCore_iOS/Source/Module/DirectMessageParticipant/LMChatMemberListViewController.swift (hunk 4 above)
Comment (pushpendralike) [reply LMChatCore_iOS/LMChatCore_iOS/Source/Module/DirectMessageParticipant/LMChatMemberListViewController.swift:36]: Done

File: LMChatCore_iOS/LMChatCore_iOS/Source/Module/DirectMessageFeed/LMChatDMFeedViewController.swift (hunk 3 above)
Comment (pushpendralike) [reply LMChatCore_iOS/LMChatCore_iOS/Source/Module/DirectMessageFeed/LMChatDMFeedViewController.swift:75]: Done

File: LMChatCore_iOS/LMChatCore_iOS/Source/Module/DirectMessageFeed/LMChatDMFeedViewController.swift (hunk 2 above)
Comment (pushpendralike) [reply LMChatCore_iOS/LMChatCore_iOS/Source/Module/DirectMessageFeed/LMChatDMFeedViewController.swift:68]: Done

File: LMChatCore_iOS/LMChatCore_iOS/Source/Module/DirectMessageFeed/LMChatDMFeedViewController.swift (hunk 1 above)
Comment (pushpendralike) [reply LMChatCore_iOS/LMChatCore_iOS/Source/Module/DirectMessageFeed/LMChatDMFeedViewController.swift:49]: Done

**Pull request merged**